### ISSUE TITLE
Questions about the baselines in the paper

### ISSUE THREAD

**Issue by JiyangZhang** [NougatCA/SPT-Code#1]
Hi,

It is a great and interesting work!
I am curious that do you have a chance to compare with the model proposed in this work [CodeT5](https://arxiv.org/abs/2109.00859
) They did evaluation on the similar datasets as yours.

Thanks.

**Comment by NougatCA** [comment]
Hello, Jiyang,

CodeT5 is an excellent work that first appeared in the arXiv in September and was published in EMNLP 2021, later than we submitted the manuscript, so we are sorry that we did not include CodeT5 as a model for comparison.

We noticed CodeT5 afterwards, but as contemporaneous work, and also for time reasons, we did not include it in subsequent versions either.

BTW, one of its pre-training tasks, Bimodal Dual Generation, a *supervised* task, makes use of the documentation and actually pre-trains CodeT5 on bimodal CodeSearchNet by means of code summarization/generation, which means it leaks the test set of downstream CodeSearchNet code summarization/generation and would lead to a very unfair comparison.

**Comment by JiyangZhang** [comment]
Hi,

Thanks for your reply. They all make sense. 

> BTW, one of its pre-training tasks, Bimodal Dual Generation, a supervised task, makes use of the documentation and actually pre-trains CodeT5 on bimodal CodeSearchNet by means of code summarization/generation, which means it leaks the test set of downstream CodeSearchNet code summarization/generation and would lead to a very unfair comparison.

The authors from CodeT5 claim they only used training set from CodeSearchNet for pretraining. As long as they did not use test set, it is fine I think. https://github.com/salesforce/CodeT5/issues/19

**Comment by NougatCA** [comment]
Thank you for providing this information, but we still think that BDG brings unfairness. Because, it makes CodeT5 utilize supervised data during pre-training, which is unfair for other models, at least ours, since we only use unlabeled data, which is also one of the contributions of our work.
